src/main/java/fr/insalyon/creatis/gasw/executor/batch/internals/BatchManager.java

```diff
@@ -0,0 +1,191 @@
+
+@Getter
+@Log4j
+public class BatchManager {
+
+    final private String            workflowId;
+    final private BatchConfig       config;
+    final private String            workingDir;
+    final private List<BatchJob>    jobs = new ArrayList<>();
+
+    private boolean initialized = false;
+    private Boolean end;
+
+    public BatchManager(final String workflowId, final BatchConfig config) {
+        this.workflowId = workflowId;
+        this.config = config;
+        this.workingDir = config.getCredentials().getWorkingDir() + workflowId;
+    }
+
+    public void init() {
+        try {
+            checkRemoteDirs();
+            checkLocalOutputsDir();
+
+            initialized = true;
+        } catch (GaswException e) {
+            log.error("Failed to init the batch manager !");
+        }
+    }
+
+    /**
+     * Create the necessary directories on the remote (batch cluster)
+     */
+    public void checkRemoteDirs() throws GaswException {
+        final List<RemoteCommand> commands = new ArrayList<>();
+
+        commands.add(new Mkdir(getWorkingDir(), ""));
+        commands.add(new Mkdir(getWorkingDir() + "/out", "-p"));
+        commands.add(new Mkdir(getWorkingDir() + "/err", "-p"));
+        commands.add(new Mkdir(getWorkingDir() + "/sh", "-p"));
+
+        for (final RemoteCommand command : commands) {
+            command.execute(config);
+
+            if (command.failed()) {
+                log.error("Failed to create the remotes dirs: " + command.getCommand());
+                throw new GaswException("Failed to create the remotes dirs: " + command.getCommand());
+            }
+        }
+    }
+
+    public void checkLocalOutputsDir() {
+        final String[] dirs = { GaswConstants.OUT_ROOT, GaswConstants.ERR_ROOT, "./cache" };
+
+        Arrays.stream(dirs)
+                .map(File::new)
+                .forEach(dir -> {
+                    if (!dir.exists()) {
+                        dir.mkdirs();
+                    }
+                });
+    }
+
+    /**
+     * Clean files created inside the workflow folder
+     */
+    public void destroy() {
+        final RemoteCommand remoteCommand = new Rm(config.getCredentials().getWorkingDir() + workflowId, "-rf");
+
+        end = true;
+        try {
+            if (remoteCommand.execute(config).failed()) {
+                log.warn("Failed to execute the remote command: " + remoteCommand.getCommand());
+            }
+        } catch (GaswException e) {
+            log.error("Failed to destroy the batch manager !");
+        }
+    }
+
+    private void submitter(final BatchJob exec) {
+        if (end == null) {
+            end = false;
+            new Thread(this.new BatchRunner()).start();
+        }
+        synchronized (this) {
+            jobs.add(exec);
+        }
+    }
+
+    public void submitter(final String jobID, final String command) {
+        final BatchJobData jobData = new BatchJobData(jobID, config);
```

Comment:
I would add `worfflowId` and `command` in the BatchJobData constructor.
Actually I would add all the stuff done here (upload /download lists etc) in BatchJobData.
I like that BatchJobData is a small data-holder class that hosts all the job configuration.

Don't forget to use the constants from GaswConstants for`./inv`, `./out` etc

Reply:
Then as this method will be almost empty, merge it with the other submitter method (to rename submitJob)